src/main/java/soot/jimple/toolkits/typing/fast/TypeResolver.java

```diff
@@ -585,9 +584,8 @@ private void split_new() {
                   deflist = defs.getDefsOfAt((Local) assign.getRightOp(), assign);
                   continue;
                 } else if (assign.getRightOp() instanceof NewExpr) {
-                  Local newlocal = jimple.newLocal("tmp", null);
```

Comment:
This local was initialized without a type. You changed that. Any idea why that was the case?

Reply:
Yes, it’s because the local generator does not accept null values for that parameter — it would throw an exception.

Reply:
Which is a good thing imho. It might not be needed in this case but users of the api should be able to expect and get a type for a local variable.